# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gkrellm/libsm.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur gkrellm-2.2.7/Makefile gkrellm-2.2.7-t2/Makefile
--- gkrellm-2.2.7/Makefile	2005-05-22 18:18:44.000000000 +0200
+++ gkrellm-2.2.7-t2/Makefile	2006-01-25 14:10:16.000000000 +0100
@@ -207,7 +207,7 @@
 	(cd po && ${MAKE} all)
 	(cd src && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -Wl,-R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm )
 	(cd server && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm" gkrellmd )
@@ -216,7 +216,7 @@
 	(cd po && ${MAKE} all)
 	(cd src && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm -pthread" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm )
 	(cd server && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm -pthread" gkrellmd )
diff -ur gkrellm-2.2.7/src/Makefile gkrellm-2.2.7-t2/src/Makefile
--- gkrellm-2.2.7/src/Makefile	2005-05-09 18:54:03.000000000 +0200
+++ gkrellm-2.2.7-t2/src/Makefile	2006-01-25 14:16:58.000000000 +0100
@@ -17,7 +17,7 @@
 MANDIRMODE ?= 755
 INSTALL ?= install
 LINK_FLAGS ?= -Wl,-E
-SMC_LIBS ?= -L/usr/X11R6/lib -lSM -lICE
+SMC_LIBS ?= $(shell pkg-config --libs sm)
 
 ifeq ($(without-gnutls),1)
 	CONFIGURE_ARGS += --without-gnutls
@@ -128,11 +128,11 @@
 
 netbsd1:
 	$(MAKE) EXTRAOBJS= SYS_LIBS="-lkvm" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -Wl,-R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm
 netbsd2:
 	$(MAKE) EXTRAOBJS= SYS_LIBS="-lkvm -pthread" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm
 
 openbsd:
